command/agent/auth/auth.go

```diff
@@ -169,12 +177,58 @@ func (ah *AuthHandler) Run(ctx context.Context, am AuthMethod) error {
 			}
 		}
 
-		secret, err := clientToUse.Logical().Write(path, data)
-		// Check errors/sanity
-		if err != nil {
-			ah.logger.Error("error authenticating", "error", err, "backoff", backoff.Seconds())
-			backoffOrQuit(ctx, backoff)
-			continue
+		var secret *api.Secret = new(api.Secret)
+		if first && ah.enableUseExistingToken {
+			var token string
+			home, err := os.UserHomeDir()
+			if err != nil {
+				return fmt.Errorf("error determining home directory: %s", err)
+			}
+			tokenPath := fmt.Sprintf("%s/.vault-token", home)
+
+			if fileExists(tokenPath) {
+				ah.logger.Debug("attempting to read preloaded token from path", "path", tokenPath)
+				tokenRaw, err := ioutil.ReadFile(tokenPath)
+				if err != nil {
+					return fmt.Errorf("error opening token file: %s", err)
+				}
+				token = string(tokenRaw)
+			} else {
+				ah.logger.Debug("attempting to read preloaded token from env VAULT_TOKEN")
+				token = os.Getenv("VAULT_TOKEN")
+			}
```

Comment:
I curious about the order here, since preferring ~/.vault-token over VAULT_TOKEN seems reversed from how the vault cli behaves. i.e. VAULT_TOKEN seems to always take precedence over a token helper like ~/.vault-token.

Reply:
No longer applicable since we now pass in the token.